mslib/msui/topview.py

```diff
@@ -312,3 +320,37 @@ def load_settings(self):
         """
         settings = load_settings_qsettings(self.settings_tag, {})
         self.getView().set_map_appearance(settings)
+
+    def make_roundtrip(self):
+        """
+        Copies the first waypoint and inserts it at the back of the list again
+        Essentially creating a roundtrip
+        """
+        # This case should never be True for users, but might be for developers at some point
+        if not self.is_roundtrip_possible():
+            return
+
+        first_waypoint = self.waypoints_model.waypoint_data(0)
+        last_waypoint = self.waypoints_model.waypoint_data(self.waypoints_model.rowCount() - 1)
+
+        self.waypoints_model.insertRows(self.waypoints_model.rowCount(), rows=1, waypoints=[
+            Waypoint(lat=first_waypoint.lat, lon=first_waypoint.lon, flightlevel=first_waypoint.flightlevel,
+                     location=first_waypoint.location)])
+
+    def is_roundtrip_possible(self):
+        """
+        Checks if there are at least 2 waypoints, and the first and last are not the same
+        """
+        condition = self.waypoints_model.rowCount() > 1
+
+        if condition:
+            first_waypoint = self.waypoints_model.waypoint_data(0)
+            last_waypoint = self.waypoints_model.waypoint_data(self.waypoints_model.rowCount() - 1)
+
+            condition = first_waypoint.lat != last_waypoint.lat or first_waypoint.lon != last_waypoint.lon or \
+                        first_waypoint.flightlevel != last_waypoint.flightlevel
+
```

Comment:
please use blank lines only if that improves readability.

Reply:
Oh. I was actually taught highly to always group my code into logical segments to make understanding it easier on first sight.

You mention this specifically on this line, is it only an issue here, or do you want no blank lines for logical segmentation at all?

Reply:
I see nothing wrong with your code. It is pep8 compliant and I maybe would have used maybe an empty line less than you. I agree with `please use blank lines only if that improves readability.` but I think that applies here. To some degree it is a matter of taste obviously.
We might talk about style on next Thursday to come to a common point of view.

Reply:
Hi, it was not meant only for that location. It is more in general, we are below 10%. That's good but in all >5K empty lines sounds much. If we compare with XML code, there is not any empty line needed. Usually many empty lines means better readablity is done by using a helper function/method. Everything is still a kind of balance. It was a hint by me too.